motionplan/motionPlanner.go

```diff
@@ -111,39 +72,74 @@ func motionPlanInternal(ctx context.Context,
 		return nil, err
 	}
 
-	logger.Infof(
+	request.Logger.Infof(
 		"planning motion for frame %s\nGoal: %v\nStarting seed map %v\n, startPose %v\n, worldstate: %v\n",
-		f.Name(),
-		frame.PoseInFrameToProtobuf(goal),
-		seedMap,
+		request.Frame.Name(),
+		frame.PoseInFrameToProtobuf(request.Goal),
+		request.Inputs,
 		spatialmath.PoseToProtobuf(startPose),
-		worldState.String(),
+		request.WorldState.String(),
 	)
-	logger.Debugf("constraint specs for this step: %v", constraintSpec)
-	logger.Debugf("motion config for this step: %v", motionConfig)
+	request.Logger.Debugf("constraint specs for this step: %v", request.ConstraintSpecs)
+	request.Logger.Debugf("motion config for this step: %v", request.Options)
 
 	rseed := defaultRandomSeed
-	if seed, ok := motionConfig["rseed"].(int); ok {
+	if seed, ok := request.Options["rseed"].(int); ok {
 		rseed = seed
 	}
-
-	sfPlanner, err := newPlanManager(sf, fs, logger, rseed)
+	sfPlanner, err := newPlanManager(sf, request.FrameSystem, request.Logger, rseed)
 	if err != nil {
 		return nil, err
 	}
-	logger.Debug("created new plan manager")
-	resultSlices, err := sfPlanner.PlanSingleWaypoint(ctx, seedMap, goal.Pose(), worldState, constraintSpec, motionConfig)
+
+	resultSlices, err := sfPlanner.PlanSingleWaypoint(
+		ctx,
+		request.Inputs,
+		request.Goal.Pose(),
+		request.WorldState,
+		request.ConstraintSpecs,
+		request.Options,
+	)
 	if err != nil {
 		return nil, err
 	}
+	plan := Plan{}
 	for _, resultSlice := range resultSlices {
 		stepMap := sf.sliceToMap(resultSlice)
-		steps = append(steps, stepMap)
+		plan = append(plan, stepMap)
 	}
+	request.Logger.Debugf("final plan steps: %s", plan.String())
+	return plan, nil
+}
 
-	logger.Debugf("final plan steps: %v", steps)
-
-	return steps, nil
+// PlanFrameMotion plans a motion to destination for a given frame with no frame system. It will create a new FS just for the plan.
+// WorldState is not supported in the absence of a real frame system.
+func PlanFrameMotion(ctx context.Context,
```

Comment:
should this also be refactored to take a request struct?

Reply:
Created RSDK-4730 to track this